src/tests/test_verification.py

```diff
@@ -1,19 +1,37 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Oxford Quantum Circuits Ltd
 
-from qat.purr.backends.verification import get_verification_model, Lucy, QPUVersion
+from qat.purr.backends.verification import get_verification_model, Lucy, QPUVersion, verify_program
+from qat.purr.compiler.config import Tket, CompilerConfig
 from qat.qat import execute
-from .qasm_utils import get_qasm2
+from .qasm_utils import get_qasm2, TestFileType, get_test_file_path
 
 import pytest
 
 
 class TestFirmwareVerificationEngines:
     def test_latest_lucy(self):
-        with pytest.raises(NotImplementedError):
-            model = get_verification_model(Lucy.Latest)
-            execute(get_qasm2("basic.qasm"), model)
+
+        model = get_verification_model(Lucy.Latest)
+        assert execute(get_qasm2("basic.qasm"), model)
 
     def test_unknown_make(self):
         model = get_verification_model(QPUVersion("something", "123"))
         assert model is None
+
+    @pytest.mark.parametrize(("input_string", "file_type", "expected_result"),
+                              [("primitives.qasm", TestFileType.QASM2, True),
+                               ("ghz.qasm", TestFileType.QASM3, True),
+                               ("ghz.qasm", TestFileType.QASM3, True),
+                               ("bell_psi_plus.ll", TestFileType.QIR, True),
+                               ("cross_ressonance.qasm", TestFileType.OPENPULSE, True),
+                               ("long_qasm.qasm", TestFileType.QASM2, False)])
+    def test_circuit_length_validation(self, input_string, file_type, expected_result):
+        program = get_test_file_path(file_type, input_string)
+
+        optim = Tket()
+        optim.disable()
+        config = CompilerConfig(optimizations=optim)
+
+        assert verify_program(program, config, QPUVersion(make="Lucy", version="latest")) == expected_result
```

Comment:
`QPUVersion.Lucy` should be fine here if memory serves. Same deal about results being try/catch.

Have two test methods one which verifies correctly one which verifies false too, can remove an argument and make this far less catch-all.

What's with the optimization disabling anyway?